### PR TITLE
MES-3398 upload rekey test

### PR DESCRIPTION
**Pull request by buzzub**
## Description and relevant Jira numbers
- Create new SendCurrentTest actions
- Prevent rekey tests from being uploaded by the `sendCompletedTestsEffect`
- Create an effect that uploads the current test and dispatches a success / fail action
- Handle the display of uploading spinner and modals on the rekey reason page using page state
- Make the upload modal handle retry scenarios by altering the wording
- Create a rekey successfully uploaded page with a link back to the journal root

## Pull Request checklist:

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist:

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop
- [x] commits are squashed

## Sign off process checklist:

- [x] Code has been tested manually
- [x] Tests are passing
- [x] PR link added to JIRA
- [x] Reviewers selected in Github
- [x] Any new reusable component/widget added to component library on Confluence

- [x] Screenshot(s) captured on the physical device (if applicable)
- [ ] Tested by QA
- [ ] PO's approval

IMG_0118

IMG_0119

IMG_0120

IMG_0122 2

### CODE REVIEW

**Review by rossfauldsbjss — APPROVED** [review]
Looks good.

File: src/modules/tests/tests.effects.ts [hunk 1]
```diff
@@ -196,9 +197,49 @@ export class TestsEffects {
...
+
+      const testToSubmit: TestToSubmit[] = [{
+        slotId,
+        index: 0,
+        payload: test,
+      }];
+
+      return this.testSubmissionProvider.submitTests(testToSubmit)
+        .pipe(
+          delay(1000),
```
Comment (ammarhaiderbjss) [inline src/modules/tests/tests.effects.ts]: What's the delay for?

Comment (buzzub) [reply src/modules/tests/tests.effects.ts]: This is gone now as it isn't needed.

**Review by szabikr — APPROVED** [review]

File: src/modules/tests/tests.effects.ts [hunk 2]
```diff
@@ -196,9 +197,48 @@ export class TestsEffects {
...
+        return of(new testActions.SendCurrentTestFailure(slotId));
+      }
+
+      const testToSubmit: TestToSubmit[] = [{
+        slotId,
+        index: 0,
+        payload: test,
+      }];
+
+      return this.testSubmissionProvider.submitTests(testToSubmit)
```
Comment (ammarhaiderbjss) [inline src/modules/tests/tests.effects.ts:223]: Looks like there might be some repetition with the `sendCompletedTestsEffect$` Effect when submitting tests. Could we look to unify them?
Could also have the added benefit of simplifying this flow, currently it has a lot going on with two levels of `switchMap`!

File: src/pages/rekey-reason/rekey-reason.effects.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      withLatestFrom(
+        this.store$.pipe(
+          select(getTests),
+          select(getCurrentTestSlotId),
+        ),
+      ),
+    )),
+    switchMap(([action, slotId]: [testActions.SendCurrentTestSuccess, string]) => {
+      return [
+        new testStatusActions.SetTestStatusCompleted(slotId),
```
Comment (ammarhaiderbjss) [inline src/pages/rekey-reason/rekey-reason.effects.ts:33]: I thought we weren't dispatching this action as there was no need?

Comment (buzzub) [reply src/pages/rekey-reason/rekey-reason.effects.ts:33]: You're totally right. Good spot.

**Pull request merged**